kafka/producer/main.go

```diff
@@ -197,21 +190,29 @@ func generateRandomNum(span int, targetRate int) int {
 	return rand.Intn((targetRate+span)-min) + min + 1
 }
 
-func loopPrint() {
-	for {
-		time.Sleep(5 * time.Second)
-		log.Printf("total Msg is: %d", totalMsg)
-		log.Printf("average burst rate is: %d", int(totalMsg)/totalBurst)
-		log.Printf("average rest duration is: %d", totalRestDur/totalBurst)
-		log.Printf("average send duration is: %d", totalSendDur/totalBurst)
+func getEnv(envName string) string {
+	value := os.Getenv(envName)
+	if value == "" {
+		fmt.Println("Environment variable " + envName + " is missing")
+		os.Exit(1)
 	}
+	return value
 }
 
-func inClusterKafkaConfig() (kafkaConfig *sarama.Config, err error) {
-	kafkaConfig = sarama.NewConfig()
-	kafkaConfig.ClientID = "kafka-on-kata-cc-mariner"
-	kafkaConfig.Producer.RequiredAcks = sarama.WaitForLocal
-	kafkaConfig.Producer.Return.Successes = true
+func getConfig() *sarama.Config {
+	config := sarama.NewConfig()
+	config.Net.DialTimeout = 10 * time.Second
+
+	config.Net.SASL.Enable = true
+	config.Net.SASL.User = "$ConnectionString"
+	config.Net.SASL.Password = getEnv(eventHubsConnStringEnvVar)
```

Comment:
We want to try to get this from AKV so that we don't have to put credentials into the ENV.